src/lib/Scenes/Collection/Collection.tsx

```diff
@@ -60,19 +67,64 @@ export class Collection extends Component<CollectionProps, CollectionState> {
           </Box>
         )
       case "collectionArtworks":
-        return <CollectionArtworks collection={this.props.collection} />
+        return (
+          <>
+            <CollectionArtworks collection={this.props.collection} />
+            <FilterModal
+              visible={this.state.isFilterArtworksModalVisible}
+              closeModal={this.handleFilterArtworksModal.bind(this)}
+            />
+          </>
+        )
       default:
         return null
     }
   }
+  onViewableItemsChanged = ({ viewableItems }) => {
```

Comment:
There's kind of a lot going on in this function, and I don't really know what it all is doing. It's possible that this is a common pattern in Emission and I just don't have experience with it, but if that's not the case, is there anything we could do to make it more clear?

Reply:
Hmm,  I don't think the objective is to always have code by pretty clear. The function takes an array of all objects that are currently in the viewport and updates the state if the conditionally rendered component we are looking to show/hide is/is not on the screen.  I'm happy to add some comments for added clarity.  Would you be interested in pairing on a re-factor?

Reply:
I actually feel the opposite - the main point of code review, IMO, is to make sure the code is clear enough that anyone without context can understand when they come across it in a few months.

I'm definitely up for pairing to refactor!

Reply:
Woops! DEFINITELY a typo, I meant _I think the objective is to always have code be pretty clear_. This is what I get for multitasking!